cmd/cluster/scale/defaulting/scaling_test.go

```diff
@@ -0,0 +1,140 @@
+package defaulting
+
+import (
+	"context"
+	"reflect"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/giantswarm/gsctl/cmd/cluster/scale/request"
+)
+
+func Test_Cmd_Cluster_Scale_Defaulting(t *testing.T) {
+	testCases := []struct {
+		Name                     string
+		AutoScalingEnabled       bool
+		CurrentScalingMax        int64
+		CurrentScalingMin        int64
+		DesiredNumWorkers        int64
+		DesiredNumWorkersChanged bool
+		DesiredScalingMax        int64
+		DesiredScalingMaxChanged bool
+		DesiredScalingMin        int64
+		DesiredScalingMinChanged bool
+		ExpectedScaling          request.Scaling
+	}{
+		{
+			Name:                     "case 0 ensures --workers-min=5 sets scaling.min to 5",
+			AutoScalingEnabled:       true,
+			CurrentScalingMax:        3,
+			CurrentScalingMin:        3,
+			DesiredNumWorkers:        0,
+			DesiredNumWorkersChanged: false,
+			DesiredScalingMax:        0,
+			DesiredScalingMaxChanged: false,
+			DesiredScalingMin:        5,
+			DesiredScalingMinChanged: true,
+			ExpectedScaling: request.Scaling{
+				Min: 5,
+				Max: 3,
```

Comment:
I assume this (min > max) is only to test that the defaulting logic touches only relevant field?

Reply:
Right, defaulting works. Validation is a different thing, which does not apply here.